libs/libarchfpga/src/physical_types_util.h

```diff
@@ -0,0 +1,132 @@
+#ifndef PHYSICAL_TYPES_UTIL_H
+#define PHYSICAL_TYPES_UTIL_H
+
+#include "physical_types.h"
+
+/********************************************************************
+ *                                                                  * 
+ *  Physical types utility functions                                *
+ *                                                                  *
+ *  This source file contains several utilities that enable the     *
+ *  interaction with the architecture's physical types.             *
+ *  Mainly, the two classes of objects accessed by the utility      *
+ *  functions in this file are the following:                       *
+ *    - physical_tile_type                                          *
+ *    - logical_block_tpye                                          *
+ *                                                                  *
+ *  All the following utilities are intended to ease the            *
+ *  developement to access the above mentioned classes and perform  *
+ *  some required operations with their data.                       *
+ *                                                                  *
+ *  Please classify such functions in this file                     *
+ *                                                                  *
+ * ******************************************************************/
+
+bool is_opin(int ipin, t_physical_tile_type_ptr type);
+bool is_input_type(t_physical_tile_type_ptr type);
```

Comment:
Should comment what these functions do.  is_input_type is an I/O pad test I think, while is_opin is a test for a single pin being an output pin of a block I think.  What index is ipin on -- a pin index on a block, or some other kind of index?)

Reply:
Done